### PR TITLE
Compiler: export function that retrieves `ERL_COMPILER_OPTIONS`

### PR DESCRIPTION
**Pull request by talentdeficit**
exposing this internal compiler functionality will save [rebar3](github.com/erlang/rebar3) from having to reimplement the function to check that beam files are up to date during compilation (see erlang/rebar3#1184)

### CODE REVIEW

File: lib/compiler/test/compile_SUITE.erl [hunk 1]
```diff
@@ -935,6 +935,23 @@ do_warnings_2([], Next, F) ->
...
+env_compiler_options(_Config) ->
+    Cases = [
+        {"bin_opt_info", [bin_opt_info]},
+        {"'S'", ['S']},
+        {"{source, \"test.erl\"}", [{source, "test.erl"}]},
+        {"[{d,macro_one,1},{d,macro_two}]", [{d, macro_one, 1}, {d, macro_two}]},
+        {"[warn_export_all, warn_export_vars]", [warn_export_all, warn_export_vars]}
+    ],
+    F = fun({Env, Expected}) ->
+        ?line true = os:putenv("ERL_COMPILER_OPTIONS", Env),
```
Comment (bjorng) [inline lib/compiler/test/compile_SUITE.erl]: Don't use the ?line macros. All uses of ?line macros have been removed from most test suites in OTP.

**Comment by bjorng** [comment]
IMO, rebar3 should not call undocumented functions. Please document the new function (also mention in the paragraph "Default Compiler Option").

Also add the information in this PR about why this functionality is needed to the commit message.

**Comment by psyeugenic** [comment]
In addition to what @bjorng said: The `?line` macros were removed for 19.0. Also, please rebase your branch to `master`.

**Comment by OTP-Maintainer** [comment]
The summary line of the commit message is too long and/or ends with a "."
Make sure the whole message follows the guidelines here: https://github.com/erlang/otp/wiki/Writing-good-commit-messages.

Bad message: Compiler: export function that retrieves `ERL_COMPILER_OPTIONS`


----------------

I am a script, I am not human 

----------------

**Comment by talentdeficit** [comment]
closing this to make pr against `master`

**Pull request closed without merging**